ompi/mca/common/ompio/common_ompio_buffer.c

```diff
@@ -39,26 +39,29 @@ static int32_t  mca_common_ompio_pagesize=4096;
 static void* mca_common_ompio_buffer_alloc_seg ( void *ctx, size_t *size );
 static void mca_common_ompio_buffer_free_seg ( void *ctx, void *buf );
 
-#if OPAL_CUDA_SUPPORT
 void mca_common_ompio_check_gpu_buf ( ompio_file_t *fh, const void *buf, int *is_gpu, 
 				      int *is_managed)
 {
-    opal_convertor_t    convertor;  
-    
+    int dev_id;
+    uint64_t flags;
+    int ret;
+
     *is_gpu=0;
     *is_managed=0;
     
-    convertor.flags=0;
-    if ( opal_cuda_check_one_buf ( (char *)buf, &convertor ) ) {
+    if (NULL == opal_accelerator.check_addr) {
```

Comment:
Why not have base functions that are essentially no-ops, so you don't need this check?  This is a performance part of the code, and we're adding unnecessary conditionals.  I'm only going to make this comment here, but it applies throughout the PR.

Reply:
Noted, will change